File: src/extractors/sites/ytdlp_extractor.cpp
#include "ytdlp_extractor.hpp"
#include "../../ytdlp_backend.hpp"
#include "../../config.hpp"

bool YtDlpExtractor::matches(
    const std::string& url
)
{
    return true;
}

std::vector<DownloadTask>
YtDlpExtractor::extract(
    const std::string& url
)
{
    DownloadTask task;

    task.url = url;

    task.use_ytdlp = true;

    task.format_id =
        Config::get_default_format();

    return { task };
}

File: android/app/src/main/java/com/mediadownloader/extractors/sites/YtDlpExtractor.kt
package com.mediadownloader.extractors.sites

import com.mediadownloader.config.Config
import com.mediadownloader.extractors.BaseExtractor
import com.mediadownloader.extractors.DownloadTask

class YtDlpExtractor : BaseExtractor {
    override fun matches(url: String): Boolean = true

    override suspend fun extract(url: String): List<DownloadTask> {
        return listOf(
            DownloadTask(
                url = url,
                useYtdlp = true,
                formatId = Config.getDefaultFormat()
            )
        )
    }
}
